src/main/java/com/minecolonies/permissions/ColonyPermissionEventHandler.java

```diff
@@ -0,0 +1,240 @@
+                  !colony.getPermissions().hasPermission(event.getEntityPlayer(), Permissions.Action.ACCESS_HUTS))
+            {
+                //todo: hopefully that works
+                event.setResult(Event.Result.DENY);
+            }
+
+        }
+    }
+
+    /**
+     * TODO Check this behavior
+     *
+     * @param event
+     */
+    @SubscribeEvent
+    public void on(final PlayerContainerEvent.Open event)
+    {
+        if (this.colony.isCoordInColony(event.getEntity().getEntityWorld(), event.getEntity().getPosition()))
+        {
+            if (this.colony.getPermissions().isColonyMember(event.getEntityPlayer()))
+            {
+                Log.getLogger().info("Colony member `" + event.getEntityPlayer().getName() + "` opens container `" + event.getEntity().getName() + "`");
+            }
+            else
+            {
+                Log.getLogger().info("Player `" + event.getEntity().getName() + "` is not a member of `" + this.colony.getName() + "`");
+            }
+        }
+        else
+        {
+            Log.getLogger().info("The opened entity is not inside `" + this.colony.getName() + "`");
+        }
+    }
+
+    /**
+     * ItemTossEvent handler.
+     *
+     * @param event ItemTossEvent
+     */
+    @SubscribeEvent
+    public void on(ItemTossEvent event)
+    {
+        final EntityPlayer playerIn = event.getPlayer();
+
+        Log.getLogger().debug(String.format("Check if playerIn `%s` is inside colony `%s`", playerIn.getName(), colony.getName()));
+        if (colony.isCoordInColony(playerIn.getEntityWorld(), playerIn.getPosition()))
+        {
+            Log.getLogger().debug(String.format("Check if playerIn `%s` has at least Rank `%s`", playerIn.getName(), colony.getName()));
+
+            Permissions.Rank rank = colony.getPermissions().getRank(playerIn);
+
+            if (rank.ordinal() < Permissions.Rank.FRIEND.ordinal()) {
+                event.setResult(Event.Result.DENY);
+            }
+        } else
+        {
+            Log.getLogger().info(String.format("Player `%s` is not inside colony `%s`", playerIn.getName(), colony.getName()));
+        }
+    }
+
+    /**
+     * Template
+     */
+    public void on()
+    {
+        //
+    }
+
+    /**
+     * This method returns TRUE if this event should be denied.
+     *
+     * @param worldIn
+     * @param posIn
+     * @param playerIn
+     * @param blockState
+     * @return
+     */
+    private boolean checkBlockEventDenied(final World worldIn, final BlockPos posIn, final EntityPlayer playerIn, final IBlockState blockState)
+    {
+        Log.getLogger().debug("Check if coordinate is in colony `" + this.colony.getName() + "`");
+        if (colony.isCoordInColony(worldIn, posIn))
+        {
+            Log.getLogger().info("Coordinate is inside colony");
+
+            Log.getLogger().debug("Check if player `" + playerIn.getName() + "` is member of `" + this.colony.getName() + "`.");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "Check if player `" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1192)